### PR TITLE
UI improvements (Show answer, Mobile)

### PR DESCRIPTION
**Pull request by reflash**
Added Fulma package for better UI components
Added a button to show correct answers (#519 )
Made changes to make the app more mobile friendly.

### CODE REVIEW

**Comment by reflash** [comment]
Here's how the new home page looks like
image

Task page display:
image

Unfortunately with a keyboard open it still isn't that nice
image

So I have an idea of how it could look like, so in case you agree I could proceed with making it so:
image
image

File: paket.dependencies [hunk 1]
```diff
@@ -20,14 +20,16 @@ group Scraper
...
-	nuget Fable.Core
+	nuget Fable.Core ~> 2.0.2
 	nuget Fable.React
-	nuget Fable.Elmish.React
-	nuget Fable.Elmish.Browser
+	nuget Fable.Elmish.React ~> 2.1
+	nuget Fable.Elmish.Browser ~> 2.1
 	nuget Fable.Elmish.Debugger
-	nuget Fable.Elmish.HMR
+	nuget Fable.Elmish.HMR ~> 2.0
```
Comment (psfinaki) [inline paket.dependencies]: Why do we have to set the specific versions of dependencies?

File: src/Client/App.fs [hunk 2]
```diff
@@ -48,62 +58,75 @@ let viewPage model dispatch =
     | VerbParticiples m ->
         VerbParticiples.view m (VerbParticiplesMsg >> dispatch)
 
+// TODO refactor this part using converter functions
```
Comment (psfinaki) [inline src/Client/App.fs]: Consider creating a task in the backlog instead of the comment :)

File: paket.dependencies (hunk 1 above)
Comment (reflash) [reply paket.dependencies]: As far as I remember it fails to compile if we use the latest version instead. Fulma library has a dependency on that one, but doesn't have a requirement for specific version. Nevertheless, it fails to work with the newest version.

File: src/Client/App.fs [hunk 3]
```diff
@@ -23,16 +27,22 @@ type Model =
     | VerbImperatives of VerbImperatives.Model
     | VerbParticiples of VerbParticiples.Model
 
+type Model = {
+    currentPage: PageModel
```
Comment (psfinaki) [inline src/Client/App.fs]: For model members we currently have PascalCase naming, or do you think we should change it?

File: paket.dependencies (hunk 1 above)
Comment (psfinaki) [reply paket.dependencies]: Alright, let it be

File: src/Client/widgets/Navbar/View.fs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    [ Navbar.Item.a [ Navbar.Item.Props [ Href "#" ] ]
+                        [ Heading.p [ Heading.Is4 ]
+                            [ str "Check Your Czech" ] ]
+                      // Make sure to have the navbar burger as the last child of the brand
+                      Navbar.burger [ Fulma.Common.CustomClass (if isBurgerOpen then "is-active" else "")
+                                      Fulma.Common.Props [
+                                        OnClick (fun _ -> dispatch ToggleBurger) ] ]
+                        [ span [ ] [ ]
+                          span [ ] [ ]
+                          span [ ] [ ] ] ]
```
[the remark this answers is not in the document]
Comment (reflash) [reply src/Client/widgets/Navbar/View.fs]: These are the actual line in the burger menu

File: src/Client/pages/Home.fs [hunk 5]
```diff
@@ -1,35 +1,35 @@
...
+open Fable.Helpers.React.Props
+open Fulma
 open Pages
 
 let view () =
     [
-        Markup.words 100 "Check Your Czech"
-        Markup.words 20 "A service to practice Czech grammar"
-        Markup.emptyLines 4
+        Markup.words "home-heading is-hidden-mobile" "Check Your Czech 🇨🇿"
```
[the remark this answers is not in the document]
Comment (reflash) [reply src/Client/pages/Home.fs]: you mean the flag emoji? It seems nice imo, but I can remove it if you think it doesn't fit

[the remark this answers is not in the document]
Comment (psfinaki) [reply src/Client/pages/Home.fs]: Hmm, that's really weird. I see the emoji when I load the desktop version on the mobile. But I don't see it from the desktop browser itself...

[the remark this answers is not in the document]
Comment (psfinaki) [reply src/Client/pages/Home.fs]: icon

[the remark this answers is not in the document]
Comment (reflash) [reply src/Client/pages/Home.fs]: Windows doesn't support flag emojis yet

[the remark this answers is not in the document]
Comment (psfinaki) [reply src/Client/pages/Home.fs]: Okay, let it be then

**Pull request merged**